doc/source/whatsnew/v0.25.0.rst

```diff
@@ -317,6 +317,35 @@ of ``object`` dtype. :attr:`Series.str` will now infer the dtype data *within* t
     s
     s.str.startswith(b'a')
 
+.. _whatsnew_0250.api_breaking.groupby_categorical:
+
+Categorical dtypes are preserved during groupby
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Previously, columns that were categorical, but not the groupby key(s) would be converted to ``object`` dtype during groupby operations. Pandas now will preserve these dtypes. (:issue:`18502`)
```

Comment:
Is this just Categorical, or any extension array?

Can we give any guidance on what operations preserve the dtype? Do aggregations?

Reply:
I *think* this *should* work for all EA, but not fully tested here, will open up a followup issue; this is only limited in the operations (e.g. filtering type of ops). see https://github.com/pandas-dev/pandas/issues/27078